app/registration/api.py

```diff
@@ -44,7 +44,39 @@ def api_register():
         data = registration.blob,
     )
 
-@registration_blueprint.route('/api/order/token')
+@registration_blueprint.route('/api/order', methods=['GET', 'POST'])
+@oauth.require_oauth('uni_order')
 @token_auth.login_required
-def validate_token():
-    return jsonify(g.uni.name)
+def api_registration_order():
+    if request.method == 'POST' \
+        and request.headers.get('Content-Type') == 'application/json':
+        req = request.get_json()
+        if not req:
+            return '', 403
+
+        new_order = len(req['order'])
+
+        for reg_id in req['confirmed']:
+            registration = Registration.query.filter_by(id = reg_id).first()
+            registration.confirmed = True
+            try:
+                registration.uni_order = req['order'].index(reg_id)
+            except ValueError:
```

Comment:
This `try`...`catch` block is confusing. May I suggest `enumerate(req['confirmed'])`?